libevmasm/Assembly.cpp

```diff
@@ -74,6 +76,218 @@ unsigned Assembly::codeSize(unsigned subTagSize) const
 	}
 }
 
+void Assembly::importAssemblyItemsFromJSON(Json::Value const& _code, vector<string> const& _sourceList)
+{
+	solAssert(m_items.empty());
+	solRequire(_code.isArray(), AssemblyImportException, "Supplied JSON is not an array.");
+	for (Json::Value const& jsonItem: _code)
+		m_items.emplace_back(createAssemblyItemFromJSON(jsonItem, _sourceList));
+
+	for (auto current = m_items.begin(); current != m_items.end(); ++current)
+	{
+		// Tags are very similar to `JUMPDEST`s. However, during the
+		// assembly json export a `JUMPDEST` is always generated after a `tag`.
+		// So we need to ignore exactly these `JUMPDEST`s during the import.
+		auto const nextIter = next(current);
+		if (
+			nextIter != m_items.end() &&
+			current->type() == AssemblyItemType::Tag &&
+			nextIter->type() == AssemblyItemType::Operation &&
+			nextIter->instruction() == Instruction::JUMPDEST
+		)
+			m_items.erase(nextIter);
+	}
+
+	// All other `JUMPDEST`'s that were not followed by a tag are treated as a fatal error.
+	for (auto const& item: m_items)
+		if (item.type() == AssemblyItemType::Operation && item.instruction() == Instruction::JUMPDEST)
+			solThrow(AssemblyImportException, "JUMPDEST instruction found that was not followed by tag.");
```

Comment:
`JUMPDEST`s are always followed by a tag: error message `JUMPDEST instruction found that was not followed by tag.` not correct.

Reply:
Tags are always followed by JUMPDESTs is what you mean in any case, don't you :-)?